docs/plugins/repository-s3.asciidoc

```diff
@@ -12,98 +12,112 @@ include::install_remove.asciidoc[]
+{
+  "type": "s3",
+  "settings": {
+    "bucket": "my_bucket"
+  }
+}
 ----
+// CONSOLE
+// TEST[skip:we don't have gcs setup while testing this]
+
 
-[[repository-s3-usage-security]]
-===== Transport security
+[[repository-s3-client]]
+==== Client Settings
 
-By default this plugin uses HTTPS for all API calls to AWS endpoints. If you wish to configure HTTP you can set
-`cloud.aws.protocol` in the elasticsearch config. You can optionally override this setting per individual service
-via: `cloud.aws.ec2.protocol` or `cloud.aws.s3.protocol`.
+The client used to connect to S3 has a number of settings available. Client settings names are of
+the form `s3.client.CLIENT_NAME.SETTING_NAME` and specified inside `elasticsearch.yml`.  The
+default client name is `default`, but can be customized with the repository setting `client`.
+For example:
 
-[source,yaml]
+[source,js]
 ----
-cloud:
-    aws:
-        protocol: https
-        s3:
-            protocol: http
-        ec2:
-            protocol: https
+PUT _snapshot/my_s3_repository
+{
+  "type": "s3",
+  "settings": {
+    "bucket": "my_bucket",
+    "client": "my_alternate_client"
+  }
+}
 ----
+// CONSOLE
+// TEST[skip:we don't have gcs setup while testing this]
 
-In addition, a proxy can be configured with the `proxy.host`, `proxy.port`, `proxy.username` and `proxy.password` settings
-(note that protocol can be `http` or `https`):
+The following settings name be specified for a client.  Some settings are sensitive and must be
+stored in the {ref}/secure-settings.html[elasticsearch keystore], and are marked as `Secure`.
 
-[source,yaml]
-----
-cloud:
-    aws:
-        protocol: https
-        proxy:
-            host: proxy1.company.com
-            port: 8083
-            username: myself
-            password: theBestPasswordEver!
-----
+`access_key`::
 
-You can also set different proxies for `ec2` and `s3`:
+    An s3 access key. Must be specified with `secret_key`. (Secure)
 
-[source,yaml]
-----
-cloud:
-    aws:
-        s3:
-            proxy:
-                host: proxy1.company.com
-                port: 8083
-                username: myself1
-                password: theBestPasswordEver1!
-        ec2:
-            proxy:
-                host: proxy2.company.com
-                port: 8083
-                username: myself2
-                password: theBestPasswordEver2!
-----
+`secret_key`::
 
-===== Read timeout
+    An s3 secret key. Must be specified with `access_key`. (Secure)
 
-Read timeout determines the amount of time to wait for data to be transferred over an established,
-open connection before the connection is timed out. Defaults to AWS SDK default value (`50s`).
-It can be configured with `cloud.aws.read_timeout` (or `cloud.aws.s3.read_timeout`) setting:
+`endpoint`::
 
-[source, yaml]
-----
-cloud.aws.read_timeout: 30s
-----
+    The s3 service endpoint to connect to. This will be automatically
+    figured out by the s3 client based on the bucket location, but
```

Comment:
I double checked and it does indeed still work. I created a bucket in `us-west-1`, and the only repository setting I set was the bucket name. I was able to create the repository, and create a snapshot just fine. I think what I have here is correct (not mentioning us-east-1), because otherwise one might be confused and think if there bucket is not in us-east-1 they would need to specify the endpoint.